From f849a9a20908949fae59bd77b485026d3884a037 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Thu, 26 Apr 2012 17:05:26 +0800
Subject: [PATCH 518/678] ENGR00180955 mx6q sabresd snvs TKT104835: fix long
 press ONOFF failed in u-boot

Need set Power Supply Glitch to 0x41736166 and clear Power Supply Glitch
Detect bit when POR or reboot or power on, otherwise system could not be
power off anymore, it will power up auto agian. These steps may be move to
ROM code or fix by soc team in the future(PDM ticket number:TKT104835),
anyway,u-boot fix the issue firsly.
Signed-off-by: Robin Gong <b38343@freescale.com>

Signed-off-by: Robin Gong <B38343@freescale.com>
---
 board/freescale/mx6q_sabresd/mx6q_sabresd.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/board/freescale/mx6q_sabresd/mx6q_sabresd.c b/board/freescale/mx6q_sabresd/mx6q_sabresd.c
index 9d227c7..8974167 100644
--- a/board/freescale/mx6q_sabresd/mx6q_sabresd.c
+++ b/board/freescale/mx6q_sabresd/mx6q_sabresd.c
@@ -736,6 +736,17 @@ void setup_splash_image(void)
 
 int board_init(void)
 {
+/* need set Power Supply Glitch to 0x41736166
+*and need clear Power supply Glitch Detect bit
+* when POR or reboot or power on Otherwise system
+*could not be power off anymore*/
+	u32 reg;
+	writel(0x41736166, SNVS_BASE_ADDR + 0x64);/*set LPPGDR*/
+	udelay(10);
+	reg = readl(SNVS_BASE_ADDR + 0x4c);
+	reg |= (1 << 3);
+	writel(reg, SNVS_BASE_ADDR + 0x4c);/*clear LPSR*/
+
 	mxc_iomux_v3_init((void *)IOMUXC_BASE_ADDR);
 	setup_boot_device();
 	fsl_set_system_rev();
-- 
1.7.7.4

